cases/deepkit.ts

```diff
@@ -0,0 +1,26 @@
+import { validate, deserialize, Negative } from '@deepkit/type';
+import { addCase } from '../benchmarks';
+
+interface DataType {
+  number: number;
+  negNumber: number & Negative;
+  maxNumber: number;
+  string: string;
+  longString: string;
+  boolean: boolean;
+  deeplyNested: {
+    foo: string;
+    num: number;
+    bool: boolean;
+  };
+}
+
+addCase('deepkit', 'parseSafe', data => {
+  return deserialize<DataType>(data);
```

Comment:
To get a deserializer for a specific type, you could do. Saves a lot of type resolving logic.

```typescript
const dataTypeDeserialize = deserializeFunction<DataType>();

addCase('deepkit', 'parseSafe', data => {
   return dataTypeDeserialize(data);
});
```

Reply:
@BasixKOR what do you think about this suggestion from marcj?

Reply:
It has to be done this way otherwise it's not a proper comparison with other cases.